app/src/vanilla/java/cm/aptoide/pt/app/view/AppViewPresenter.java

```diff
@@ -0,0 +1,619 @@
+        .doOnNext(app -> appViewNavigator.navigateToStore(app.getDetailedApp()
+            .getStore()))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnFollowStore() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickFollowStore())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .observeOn(viewScheduler)
+        .flatMapCompletable(model -> {
+          if (model.isStoreFollowed()) {
+            view.setFollowButton(true);
+            appViewAnalytics.sendOpenStoreEvent();
+            appViewNavigator.navigateToStore(model.getDetailedApp()
+                .getStore());
+            return Completable.complete();
+          } else {
+            view.setFollowButton(false);
+            appViewAnalytics.sendFollowStoreEvent();
+            view.displayStoreFollowedSnack(model.getDetailedApp()
+                .getStore()
+                .getName());
+            return appViewManager.subscribeStore(model.getDetailedApp()
+                .getStore()
+                .getName());
+          }
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> {
+          throw new OnErrorNotImplementedException(err);
+        });
+  }
+
+  private void handleClickOnOtherVersions() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickOtherVersions())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .doOnNext(model -> {
+          appViewAnalytics.sendOtherVersionsEvent();
+          appViewNavigator.navigateToOtherVersions(model.getDetailedApp()
+              .getName(), model.getDetailedApp()
+              .getIcon(), model.getDetailedApp()
+              .getPackageName());
+        })
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnTrustedBadge() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> view.clickTrustedBadge())
+        .flatMapSingle(__ -> appViewManager.getDetailedAppViewModel(appId, packageName))
+        .doOnNext(model -> view.showTrustedDialog(model.getDetailedApp()))
+        .compose(view.bindUntilEvent(View.LifecycleEvent.DESTROY))
+        .subscribe(__ -> {
+        }, err -> crashReport.log(err));
+  }
+
+  private void handleClickOnRateApp() {
+    view.getLifecycle()
+        .filter(event -> event.equals(View.LifecycleEvent.CREATE))
+        .flatMap(__ -> Observable.merge(view.clickRateApp(), view.clickRateAppLarge(),
+            view.clickRateAppLayout()))
```

Comment:
something I was playing with - but not required to change:

I think this Observable merge can be replaced by :
.flatMap(created -> view.clickRateApp())
.mergeWith(view.clickRateAppLarge())
.mergeWith(view.clickRateAppLayout())